one-ui/src/main/java/com/marklogic/hub/curation/controllers/AbstractArtifactController.java

```diff
@@ -1,49 +1,55 @@
+    protected ResponseEntity<ArrayNode> getArtifacts() {
+        return new ResponseEntity<>(artifactManager.getArtifacts(this.getArtifactType()), HttpStatus.OK);
     }
 
-    protected ResponseEntity<JsonNode> getArtifact(String artifactName) {
-        return new ResponseEntity<>(getArtifactService().getArtifact(this.getArtifactType(), artifactName), HttpStatus.OK);
+    protected ResponseEntity<ObjectNode> updateArtifact(String artifactName, JsonNode artifactJson) {
+        return new ResponseEntity<>(artifactManager.updateArtifact(this.getArtifactType(), artifactName, artifactJson), HttpStatus.OK);
     }
 
-    protected ResponseEntity<JsonNode> deleteArtifact(String artifactName) {
-        return new ResponseEntity<>(getArtifactService().deleteArtifact(this.getArtifactType(), artifactName), HttpStatus.OK);
+    protected ResponseEntity<ObjectNode> getArtifact(String artifactName) {
+        return new ResponseEntity<>(artifactManager.getArtifact(this.getArtifactType(), artifactName), HttpStatus.OK);
     }
 
-    protected ResponseEntity<JsonNode> validateArtifact(String artifactName, JsonNode artifactJson) {
-        return new ResponseEntity<>(getArtifactService().validateArtifact(this.getArtifactType(), artifactName, artifactJson), HttpStatus.OK);
+    protected void deleteArtifact(String artifactName) throws IOException {
+       artifactManager.deleteArtifact(this.getArtifactType(), artifactName);
     }
 
-    protected ArtifactService getArtifactService() {
-        DatabaseClient dataServicesClient = hubConfig.newStagingClient(null);
-        return ArtifactService.on(dataServicesClient);
+    protected ResponseEntity<ObjectNode> validateArtifact(String artifactName, JsonNode artifactJson) {
+        return new ResponseEntity<>(artifactManager.validateArtifact(this.getArtifactType(), artifactName, artifactJson), HttpStatus.OK);
     }
 
-    protected ResponseEntity<JsonNode> getArtifactSettings(String artifactName) {
-        return new ResponseEntity<>(getArtifactService().getArtifactSettings(this.getArtifactType(), artifactName), HttpStatus.OK);
+    protected ResponseEntity<ObjectNode> getArtifactSettings(String artifactName) {
+        return new ResponseEntity<>(artifactManager.getArtifactSettings(this.getArtifactType(), artifactName), HttpStatus.OK);
     }
 
-    protected ResponseEntity<JsonNode> updateArtifactSettings(String artifactName, JsonNode settings) {
-        return new ResponseEntity<>(getArtifactService().setArtifactSettings(this.getArtifactType(), artifactName, settings), HttpStatus.OK);
+    protected ResponseEntity<ObjectNode> updateArtifactSettings(String artifactName, JsonNode settings) {
+        return new ResponseEntity<>(artifactManager.updateArtifactSettings(this.getArtifactType(), artifactName, settings), HttpStatus.OK);
     }
 
     protected abstract String getArtifactType();
+
+    public void afterPropertiesSet() {
+        this.artifactManager = new ArtifactManagerImpl(this.hubConfig);
```

Comment:
Because subclasses of this class are Spring Controller beans, and thus managed as singletons, I think this still runs into a problem where there's a single ArtifactManager tied to... I guess the first user that hits this? That's my guess - that if a second user logs in, that user will end up using an ArtifactManager that is tied to the first user's HubConfig. I'm not certain though.

I think the safer approach is just to instantiate an ArtifactManagerImpl when you need it - i.e. have a "protected ArtifactManager newArtifactManager()" method that calls "new ArtifactManagerImpl(this.hubConfig)". That ensures you get an ArtifactManager tied to the current user session.

Reply:
@rjrudin I believe that the HubConfig has a separate instance for a different session.

Reply:
HubConfigSession is SessionScoped. If we couldn't trust `this.artifactManager` to not be shared between sessions, I'm not sure how we could trust `this.hubConfig` to not be shared across sessions.

Reply:
I think it's worth testing, just to verify that two users don't end up using the same HubConfig. 

I'm mostly wondering if the scoping is "lost" somehow when it's used to construct another object.